pkg/component/autoscaling/vpa/admissioncontroller.go

```diff
@@ -140,21 +145,40 @@ func (v *vpa) reconcileAdmissionControllerService(service *corev1.Service) {
 	switch v.values.ClusterType {
 	case component.ClusterTypeSeed:
 		metav1.SetMetaDataAnnotation(&service.ObjectMeta, resourcesv1alpha1.NetworkingFromWorldToPorts, fmt.Sprintf(`[{"protocol":"TCP","port":%d}]`, vpaconstants.AdmissionControllerPort))
+		utilruntime.Must(gardenerutils.InjectNetworkPolicyAnnotationsForSeedScrapeTargets(service, networkingv1.NetworkPolicyPort{
+			Port:     ptr.To(intstr.FromInt32(admissionControllerMetricsPort)),
+			Protocol: ptr.To(corev1.ProtocolTCP),
+		}))
 	case component.ClusterTypeShoot:
 		utilruntime.Must(gardenerutils.InjectNetworkPolicyAnnotationsForWebhookTargets(service, networkingv1.NetworkPolicyPort{
 			Port:     ptr.To(intstr.FromInt32(vpaconstants.AdmissionControllerPort)),
 			Protocol: ptr.To(corev1.ProtocolTCP),
 		}))
+		utilruntime.Must(gardenerutils.InjectNetworkPolicyAnnotationsForScrapeTargets(service, networkingv1.NetworkPolicyPort{
+			Port:     ptr.To(intstr.FromInt32(admissionControllerMetricsPort)),
+			Protocol: ptr.To(corev1.ProtocolTCP),
+		}))
+		metav1.SetMetaDataAnnotation(&service.ObjectMeta, resourcesv1alpha1.NetworkingPodLabelSelectorNamespaceAlias, v1beta1constants.LabelNetworkPolicyShootNamespaceAlias)
 	}
 
 	service.Spec.Selector = getAppLabel(admissionController)
 	desiredPorts := []corev1.ServicePort{
 		{
+			Name:       metricsPortName,
+			Port:       admissionControllerMetricsPort,
+			Protocol:   corev1.ProtocolTCP,
+			TargetPort: intstr.FromInt32(admissionControllerMetricsPort),
+		},
+		{
+			Name:       serverPortName,
 			Port:       admissionControllerServicePort,
 			TargetPort: intstr.FromInt32(vpaconstants.AdmissionControllerPort),
 		},
 	}
 	service.Spec.Ports = kubernetesutils.ReconcileServicePorts(service.Spec.Ports, desiredPorts, "")
+	for label, value := range getAllLabels(admissionController) {
+		metav1.SetMetaDataLabel(&service.ObjectMeta, label, value)
+	}
```

Comment:
Nit: I would move it on top of the func to have metadata updates first, spec updates - later.

```diff
diff --git a/pkg/component/autoscaling/vpa/admissioncontroller.go b/pkg/component/autoscaling/vpa/admissioncontroller.go
index fcd544a1c..f34cd6b4d 100644
--- a/pkg/component/autoscaling/vpa/admissioncontroller.go
+++ b/pkg/component/autoscaling/vpa/admissioncontroller.go
@@ -139,6 +139,10 @@ func (v *vpa) reconcileAdmissionControllerClusterRoleBinding(clusterRoleBinding
 }
 
 func (v *vpa) reconcileAdmissionControllerService(service *corev1.Service) {
+	for label, value := range getAllLabels(admissionController) {
+		metav1.SetMetaDataLabel(&service.ObjectMeta, label, value)
+	}
+
 	topologAwareRoutingEnabled := v.values.AdmissionController.TopologyAwareRoutingEnabled && v.values.ClusterType == component.ClusterTypeShoot
 	gardenerutils.ReconcileTopologyAwareRoutingMetadata(service, topologAwareRoutingEnabled, v.values.RuntimeKubernetesVersion)
 
@@ -176,9 +180,6 @@ func (v *vpa) reconcileAdmissionControllerService(service *corev1.Service) {
 		},
 	}
 	service.Spec.Ports = kubernetesutils.ReconcileServicePorts(service.Spec.Ports, desiredPorts, "")
-	for label, value := range getAllLabels(admissionController) {
-		metav1.SetMetaDataLabel(&service.ObjectMeta, label, value)
-	}
 }
 
 func (v *vpa) reconcileAdmissionControllerDeployment(deployment *appsv1.Deployment, serviceAccountName *string) {
```

Reply:
done.